### DISCUSSION TITLE
Part-DB 1.13.0 - PostgreSQL support, natural sorting and more

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#638] [Announcements]
# Part-DB 1.13.0 - PostgreSQL support, natural sorting and more

**If you like Part-DB, please consider donating to help further development. See the sponsor button on the [homepage](https://github.com/Part-DB/Part-DB-server).**

## PostgreSQL

Part-DB now supports [PostgreSQL](https://www.postgresql.org/) as database. See [documentation](https://docs.part-db.de/installation/choosing_database.html) on how it compares with other database types.

## Natural Sorting

Part-DB now supports natural sorting in part tables and tress.

So instead of sorting elements like `DIP-14`, `DIP-28`, `DIP-4` , `DIP-8`, the elements now get sorted like expected as `DIP-4`, `DIP-8`, `DIP-14` and `DIP-28`.

If you are using PostgreSQL or MariaDB 10.7+ then this kind of sorting is supported natively and natural sorting gets automatically enabled.

If you are running SQLite or MySQL then it is possible to emulate natural sorting by setting the `DATABASE_EMULATE_NATURAL_SORT` to 1.

Please note that this emulation can be very slow and might have some issues and quirks. 

# Full changelog

## New features
* Support for PostgreSQL as database
* Support for natural sorting of tables and trees, on MariaDB 10.7 and PostgreSQL. On other platforms, a (slow) emulation is possible. See [here](https://docs.part-db.de/installation/choosing_database.html#natural-sorting) for more information
* Support for importing parameters and attachments
* Added scan button to navbar in mobile view (PR #626, thanks @RaptorDE)

## Bugfixes
* Show correct data in partUnit column in part tables (fixes #630)
* Fixed error message, when using datetime filters with a range
* Use correct frontend translations, when using a locale with subregion (see #563)
* Fixed `--overwrite` option of backup command not working (issue #635)
* Hide tooltip after clicking navbar visibility toggler (#628)
* Hide mouser parts with an invalid product id (#616)
* Handle export of recursive datastructures better
* Fixed some issues with the timetravel feature

## Miscellaneous
* Updated dependencies
* Fixed some deprecations
* Improvements of code quality

## New Contributors
* @RaptorDE made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/626

**Full Changelog**: https://github.com/Part-DB/Part-DB-server/compare/v1.12.1...v1.13.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Part-DB/Part-DB-server/releases/tag/v1.13.0'>Part-DB 1.13.0</a>.</em>

**Comment by amitzioni** [comment]
Thank you very much for the new version. 
I installed part-db by downloading the archives and unpacking them. However now I see that for the update I should do `git pull`. Is there any other mechanism for updating part-db ?

**Comment by RaptorDE** [comment]
@jbtronics thanks for the update, could you please update the example csv and documentation in the wiki for the new import functions?

**Comment by TIGER53501** [comment]
Hello, how to configure docker-compose.yaml for use Postgre?

**Comment by Tropaion** [comment]
@jbtronics Does sorting by prefix also work with natural sorting? For example R, kR, MR?
